lib/Service/RoomService.php

```diff
@@ -851,4 +851,24 @@ public function deleteRoom(Room $room): void {
 			));
 		}
 	}
+
+	public function setCanMentionEveryone(Room $room, int $canMentionEveryone): void {
+		if (!in_array($canMentionEveryone, [0, 1])) {
+			throw new InvalidArgumentException('config');
+		}
+
+		$old = $room->getCanMentionEveryone();
```

Comment:
check if old === new before dispatching events and running a query on DB like on `setDescription`